src/main/java/core/di/beans/factory/DefaultMethodInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package core.di.beans.factory;
+
+import core.di.beans.factory.aop.Advice;
+import core.di.beans.factory.aop.MethodMatcher;
+import core.di.beans.factory.aop.PointCut;
+import net.sf.cglib.proxy.MethodInterceptor;
+import net.sf.cglib.proxy.MethodProxy;
+
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+public class DefaultMethodInterceptor implements MethodInterceptor {
+
+    private final Advice advice;
+    private final MethodMatcher methodMatcher;
+
+    public DefaultMethodInterceptor(Advice advice, PointCut pointCut) {
+        this.advice = advice;
+        final Optional<PointCut> maybePointCut = Optional.ofNullable(pointCut);
+        if (maybePointCut.isPresent()) {
+            this.methodMatcher = maybePointCut.get().getMethodMatcher();
+        } else {
+            this.methodMatcher = (method, targetClass) -> true;
+        }
```

Comment:
`Optional` 을 활용한다면 아래와 같이 사용할 수도 있겠네요 :)
```suggestion
        this.methodMatcher = Optional.ofNullable(pointCut)
                .map(PointCut::getMethodMatcher)
                .orElse((method, targetClass) -> true);
```